client/sw/register.spec.js

```diff
@@ -70,6 +70,23 @@ test('sw: register: registerSW: http', async (t, {location, navigator}) => {
     t.end();
 });
 
+test('sw: register: registerSW: https self-signed', async (t, {location, navigator}) => {
+    Object.assign(location, {
+        protocol: 'https',
+        hostname: 'self-signed.badssl.com',
+    });
+    
+    const {register} = navigator.serviceWorker;
```

Comment:
```suggestion
    const {register} = navigator.serviceWorker;
    register.throws(Error('Cannot register service worker!'));
```

https://github.com/cloudcmd/stub#stubthrowserror